src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            return (List<T>) bulldozerProducer.get();
+        }
+        if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            return (List<T>) excavatorProducer.get();
+        }
+        if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            return (List<T>) truckProducer.get();
+        }
+        return List.of();
```

Comment:
Keep if-else constructions simple. After each return statement within an if clause, an else is not necessary. Consider removing unnecessary else clauses to make the code cleaner.